postgres/src/main/java/org/jdbi/v3/postgres/PostgresTypes.java

```diff
@@ -48,11 +50,21 @@ public static String getTypeName(Class clazz) {
      *
      * @param connection connection on which to add all registered PostgreSQL custom types
      */
-    public static void addTypesToConnection(PGConnection connection) throws SQLException {
-        Iterator it = TYPES.entrySet().iterator();
-        while (it.hasNext()) {
-            Map.Entry pair = (Map.Entry) it.next();
-            connection.addDataType((String) pair.getValue(), (Class) pair.getKey());
+    public static void addTypesToConnection(PGConnection connection) {
+        TYPES.forEach((clazz, type) -> addTypeToConnection(connection, type, clazz));
+    }
+
+    /**
+     * Add handler for PostgreSQL custom type
+     *
+     * @param connection connection on which to add PostgreSQL custom types
+     * @throws java.sql.SQLException
+     */
+    private static void addTypeToConnection(PGConnection connection, String type, Class<? extends PGobject> clazz) {
+        try {
+            connection.addDataType(type, clazz);
+        } catch (SQLException ex) {
+            LOG.error("Error while adding handler for PostgreSQL's custom type", ex);
```

Comment:
Did you write this method and add this logger just to swallow the exception so you could use the modern iteration? I'd like to point out two general bad practices in this idea. First, swallowing exceptions by only logging them when those exceptions aren't supposed to occur is sweeping things under the rug. Programs exploding when something goes wrong is a very, very good thing. Secondly, if you have a case like this where a checked exception gets in your way because it can't be thrown, the usual solution is to wrap it in a RuntimeException. In Jdbi we have a tool that mirrors `jool`, the `Unchecked` class, which will let you hide the exception without sweeping it under the rug or adding misinformative code:

`TYPES.forEach((clazz, type) -> Unchecked.biConsumer(connection::addDataType).accept(type, clazz));`